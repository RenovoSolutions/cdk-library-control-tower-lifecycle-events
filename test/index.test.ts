import { Stack, App } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import {
  CreatedAccountRule,
  CreatedAccountByOrganizationsRule,
  UpdatedManagedAccountRule,
} from '../src/index';

test('Snapshot', () => {
  const app = new App();
  const stack = new Stack(app, 'TestStack');

  new CreatedAccountRule(stack, 'CreatedAccountRule', {});
  new CreatedAccountByOrganizationsRule(stack, 'CreatedAccountByOrganizationsRule', {});
  new UpdatedManagedAccountRule(stack, 'UpdatedManagedAccountRuleDetailed', {
    accountId: '123456789012',
    accountName: 'ExampleAccount',
    ouId: 'o-123456789012',
    ouName: 'ExampleOrganizationalUnit',
  });
  new UpdatedManagedAccountRule(stack, 'UpdatedManagedAccountRuleLessDetailed', {
    accountName: 'ExampleAccount',
    ouName: 'ExampleOrganizationalUnit',
  });
  new UpdatedManagedAccountRule(stack, 'UpdatedManagedAccountRule', {});

  expect(Template.fromStack(stack)).toMatchSnapshot();
});