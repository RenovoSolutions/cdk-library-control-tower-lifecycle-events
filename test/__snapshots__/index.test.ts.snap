// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Snapshot 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "CreatedAccountByOrganizationsRuleDAEAFAC8": {
      "Properties": {
        "Description": "A rule for new account creation in Organizations.",
        "EventPattern": {
          "detail": {
            "eventName": [
              "CreateAccountResult",
            ],
            "serviceEventDetails": {
              "createAccountStatus": {
                "state": [
                  "SUCCEEDED",
                ],
              },
            },
          },
          "detail-type": [
            "AWS Service Event via CloudTrail",
          ],
          "source": [
            "aws.organizations",
          ],
        },
        "State": "ENABLED",
      },
      "Type": "AWS::Events::Rule",
    },
    "CreatedAccountRuleF5BFAAF7": {
      "Properties": {
        "Description": "A rule for new account creation in Control Tower.",
        "EventPattern": {
          "detail": {
            "eventName": [
              "CreateManagedAccount",
            ],
            "serviceEventDetails": {
              "createManagedAccountStatus": {
                "state": [
                  "SUCCEEDED",
                ],
              },
            },
          },
          "detail-type": [
            "AWS Service Event via CloudTrail",
          ],
          "source": [
            "aws.controltower",
          ],
        },
        "State": "ENABLED",
      },
      "Type": "AWS::Events::Rule",
    },
    "UpdatedManagedAccountRule7A7C9415": {
      "Properties": {
        "Description": "A rule for updated accounts managed by Control Tower.",
        "EventPattern": {
          "detail": {
            "eventName": [
              "UpdateManagedAccount",
            ],
            "serviceEventDetails": {
              "updateManagedAccountStatus": {
                "state": [
                  "SUCCEEDED",
                ],
              },
            },
          },
          "detail-type": [
            "AWS Service Event via CloudTrail",
          ],
          "source": [
            "aws.organizations",
          ],
        },
        "State": "ENABLED",
      },
      "Type": "AWS::Events::Rule",
    },
    "UpdatedManagedAccountRuleDetailed4A80DF35": {
      "Properties": {
        "Description": "A rule for updated accounts managed by Control Tower.",
        "EventPattern": {
          "detail": {
            "eventName": [
              "UpdateManagedAccount",
            ],
            "serviceEventDetails": {
              "updateManagedAccountStatus": {
                "account": {
                  "accountId": "123456789012",
                  "accountName": "ExampleAccount",
                },
                "organizationalUnit": {
                  "organizationalUnitId": "o-123456789012",
                  "organizationalUnitName": "ExampleOrganizationalUnit",
                },
                "state": [
                  "SUCCEEDED",
                ],
              },
            },
          },
          "detail-type": [
            "AWS Service Event via CloudTrail",
          ],
          "source": [
            "aws.organizations",
          ],
        },
        "State": "ENABLED",
      },
      "Type": "AWS::Events::Rule",
    },
    "UpdatedManagedAccountRuleLessDetailed613FEC73": {
      "Properties": {
        "Description": "A rule for updated accounts managed by Control Tower.",
        "EventPattern": {
          "detail": {
            "eventName": [
              "UpdateManagedAccount",
            ],
            "serviceEventDetails": {
              "updateManagedAccountStatus": {
                "account": {
                  "accountName": "ExampleAccount",
                },
                "organizationalUnit": {
                  "organizationalUnitName": "ExampleOrganizationalUnit",
                },
                "state": [
                  "SUCCEEDED",
                ],
              },
            },
          },
          "detail-type": [
            "AWS Service Event via CloudTrail",
          ],
          "source": [
            "aws.organizations",
          ],
        },
        "State": "ENABLED",
      },
      "Type": "AWS::Events::Rule",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
